receiver/postgresqlreceiver/config.go

```diff
@@ -0,0 +1,98 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresqlreceiver
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	Username                                string   `mapstructure:"username"`
+	Password                                string   `mapstructure:"password"`
+	Databases                               []string `mapstructure:"databases"`
+	Host                                    string   `mapstructure:"host"`
+	Port                                    int      `mapstructure:"port"`
+	SSLConfig                               `mapstructure:",squash"`
+}
+
+type SSLConfig struct {
+	SSLMode     string `mapstructure:"ssl_mode"`
+	SSLRootCert string `mapstructure:"ssl_root_cert"`
+	SSLCert     string `mapstructure:"ssl_cert"`
+	SSLKey      string `mapstructure:"ssl_key"`
+}
```

Comment:
There is a package in the base collector that provides TLS settings, so typically just embad `configtls.TLSClientSettings`. Here's an [example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e066c9cdd3ebc1fd288e1e74479cc3f6fa9caa61/receiver/redisreceiver/config.go#L35). Besides providing some useful functionality, this also helps standardize configuration across receivers.

Reply:
The lib/pq module has very specific TLS config and it's not compatible with some of the settings in configtls.TLSClientSettings (Setting TLS version support is not available, overriding ServerName is not available). It uses crypto/tls under the hood but it doesn't allow direct configuration as it tries to mimic postgres's libpq configuration. 

Is standardizing on TLSClientSettings the desired behavior in this case when it's not completely compatible with the config?

Reply:
Similar to above, we should look for ways to use the standard config if possible. A couple thoughts:
- We don't necessarily need to use the functions provided by the config struct. If they are helpful, great. But if not, we can just pull the info off the values provided.
- We can ignore/disable fields in the `TLSClientSettings` that are not helpful. At a minimum, we can just omit them from the documentation. If necessary or if helpful to the user, we can enforce that they are not used (or not changed from a default value) in config validation.

Maybe there are other considerations, but can you give it another look from this perspective?